File: 53-maximum-subarray/53-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
       if(nums.length == 1){
           return nums[0];
       }
        int posmax = 0;
        int max = Integer.MIN_VALUE;
        for(int i=0;i<nums.length;i++){
        
                 posmax += nums[i];
                
                if(posmax > max){
                    max = posmax;
                }  
            if(posmax<0){
                posmax=0;
            }
                
            }
        
        return max;
        
    }
}

File: 53-maximum-subarray/53-maximum-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    if(nums.length == 1){
        return nums[0];
    }
    
     let posmax = nums[0];
    let max = nums[0];
   
    for(let i=1;i<nums.length;i++){
        
        posmax = Math.max(nums[i],posmax+nums[i]);
        
        max = Math.max(posmax,max);
       
         
        
        
    }
    return max;
    
    
};